pkg/distro/fedora/images.go

```diff
@@ -234,6 +234,10 @@ func osCustomizations(
 		osc.CACerts = ca.PEMCerts
 	}
 
+	if imageConfig.MachineIdUninitialized != nil {
+		osc.MachineIdUninitialized = *imageConfig.MachineIdUninitialized
```

Comment:
Do we need `mageConfig.MachineIdUninitialized` to be a pointer? Given that `osc.MachineIdUninitialized ` is a bool and always defaults to false it seems `imageConfig.MachineIdUninitialized` could also just be a bool (not a pointer)? This would also resolve the confusion of having a var with the same name in one place that is a bool and another that is a *bool

Reply:
I don't know. I kept the consistent idiom of pointer-to in `ImageConfig` and direct in `OSCustomizations` which was already there for other `bool` and `string` types.

Reply:
@thozza My main point specific to this is that everything that is not a `*T` in OSCustomizations does not need to be a `*T` in ImageConfig (unless I miss something of course). I.e. in OSCustomizations we have some bools (like NoBLS) and some *bool (like SELinuxForceRelabel). And for cases like "NoBLS" (and now MachineIdUninitialized) I don't see why they need to be *bool if they are collapsed into a bool anyway.

Unrelated to the above I did  https://github.com/osbuild/images/pull/860 for the points vs type thing a while ago, it never got traction so I closed it (and its value can be debated, I personally like it but I see the argument that this is just unneeded abstraction).

Reply:
I'm fine with either of these approaches but I'd like to do it as a follow-up so I can take a look at all the properties in `ImageConfig` + `OSCustomizations` to see which ones need, and need not be nullable.

Reply:
> I don't see why they need to be *bool if they are collapsed into a bool anyway.

The unset value is treated differently from the default (`0`, `false`, etc) when merging image configs in `InheritFrom()`.  In other words,
```
<image config with Uninitialized=nil>.InheritFrom(<image config with Uninitialized=default>)
```
with return an image config with `Uninitialized=default`.

```
<image config with Uninitialized=&false>.InheritFrom(<image config with Uninitialized=default>)
```
will return an image config with `Uninitialized=&false`.

Reply:
Thanks, the InheritFrom() mechanism was the missing piece for me, this is obviously how it should be. Sorry for the noise!

Reply:
Sorry, the link in my first response pointed to a comment inside the `InheritFrom()` method, but it was probably not so obvious what I meant by that.

We never merge or inherit `OSCustomizations`, so we don't need to use pointers there.